### PR TITLE
Support latest tox v4

### PR DESCRIPTION
**Pull request by kclowes**
### What was wrong?
A tox upgrade broke our lint CI runs. 

### How was it fixed?
Added an upper pin to the tox install in our CI config

### Todo:
- [x] Add entry to the [release notes](https://github.com/ethereum/web3.py/blob/master/newsfragments/README.md)

#### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

**Review by pacrob — APPROVED** [review]
lgtm!

File: .circleci/config.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ common: &common
         command: git submodule update --init --recursive
     - run:
         name: install dependencies
-        command: pip install --user tox
+        command: pip install --user tox==3.*
```
Comment (fselmo) [inline .circleci/config.yml:17]: This adds a bit more surface area when updating the tox version. I think that's fine but we should maybe add a comment in `setup.py` to update this as well when we fix the breaking changes. Thoughts?

**Review by fselmo — APPROVED** [review]
~~lgtm, just left one nit that might help us track down the different places where we may need to update tox.~~ [outdated]

File: .circleci/config.yml (hunk 1 above)
Comment (kclowes) [reply .circleci/config.yml:17]: good call. Will add a comment!

**Review by fselmo — APPROVED** [review]
I prefer this route! Consider this re-approved 🚀 ... just needs the newsfragment message updated

**Comment by kclowes** [comment]
You're so fast! Thanks!

**Pull request merged**